include/dlaf/matrix/panel.h

```diff
@@ -0,0 +1,479 @@
+    auto layout = tileLayout(dist);
+    return {std::move(dist), layout};
+  }
+
+  /// Create a Panel related to the Matrix passed as parameter.
+  ///
+  /// The Panel is strictly related to its parent dlaf::Matrix.
+  /// In particular, it will create a Row or Column with the same size of its parent matrix (local),
+  /// considering the specified offset from the top left origin.
+  ///
+  /// e.g. a 4x5 matrix with an offset 2x1 will have either:
+  /// - a Panel<Col> 2x1
+  /// - or a Panel<Row> 4x1
+  Panel(matrix::Distribution dist_matrix, LocalTileSize start)
+      : data_(setupMatrix(dist_matrix, start)), dist_matrix_(dist_matrix),
+        bias_(start.get(component(axis))) {
+    DLAF_ASSERT_HEAVY(data_.nrTiles().get(axis) == 1, data_.nrTiles());
+
+    const LocalTileSize end = dist_matrix_.localNrTiles();
+    setRange(start, end);
+
+    external_.resize(data_.nrTiles().get(component(axis)));
+
+    DLAF_ASSERT_HEAVY(data_.distribution().localNrTiles().linear_size() == external_.size(),
+                      data_.distribution().localNrTiles().linear_size(), external_.size());
+  }
+
+  /// Given a matrix index, compute the internal linear index
+  SizeType linearIndex(const LocalTileIndex& index) const noexcept {
+    const auto idx = index.get(component(axis));
+
+    DLAF_ASSERT_MODERATE(idx >= start_, idx, start_);
+
+    return idx - bias_;
+  }
+
+  /// Given a matrix index, compute the projected internal index
+  ///
+  /// It is similar to what linear_index does, so it takes into account the @p start_,
+  /// but it computes a 2D index instead of a linear one.
+  /// The 2D index is the projection of the given index, i.e. in a Panel<Col> the Col for index
+  /// will always be 0 (and relatively for a Panel<Row>)
+  LocalTileIndex fullIndex(LocalTileIndex index) const {
+    index = LocalTileIndex(component(axis), linearIndex(index));
+
+    DLAF_ASSERT_HEAVY(index.isIn(LocalTileSize(component(axis), end_, 1)), index,
+                      LocalTileSize(component(axis), end_, 1));
+
+    return index;
+  }
+
+  /// Given a matrix index, check if the corresponding tile in the panel is external or not
+  bool isExternal(const LocalTileIndex idx_matrix) const noexcept {
+    return external_[linearIndex(idx_matrix)].valid();
+  }
+
+  ///> Local matrix used for storing the panel data
+  matrix::Matrix<T, D> data_;
+
+  ///> Parent matrix which this panel is related to
+  Distribution dist_matrix_;
+
+  ///> It represents from where it is necessary to allocate memory (fixed at construction time)
+  SizeType bias_;
+  ///> It represents from where the panel gives access to tiles
+  SizeType start_;
+  ///> It represents the last
+  SizeType end_;
+
+  ///> Container for references to external tiles
+  common::internal::vector<hpx::shared_future<ConstTileType>> external_;
+  ///> Keep track of usage status of internal tiles (accessed or not)
+  std::set<SizeType> internal_;
+};
+
+template <Coord axis, class T, Device device>
```

Comment:
Do we need `axis` to be a compile-time parameter? It seems to only be used in runtime context.

Reply:
Is your comment related to this?
https://github.com/eth-cscs/DLA-Future/pull/309#discussion_r625544384

Reply:
Argh, yes, I thought my previous comment was lost. Thanks for the reply, I will look into it!